web/client/epics/maps.js

```diff
@@ -295,7 +295,12 @@ const getMapsResourcesByCategoryEpic = (action$, store) =>
                     .switchMap((response) => getContextNames(response).switchMap(responseWithContext => {
                         // category is required to identify maps for detail card. It's missing from this entry point
                         return Rx.Observable.of(
-                            mapsLoaded({...responseWithContext, results: responseWithContext?.results?.map(res => ({...res, category: {name: "MAP"}}))}, opts.params, searchText)
+                            mapsLoaded({
+                                ...responseWithContext,
+                                results: isArray(responseWithContext.results) ?
```

Comment:
responseWithContext.results is "" in case of empty results. I suggest to use [] as default, to avoid other errors in the future.

Reply:
@offtherailz it should be noted that the server in case of empty results returns an empty string for some reason which is not very sensible, and probably should also be fixed.

Reply:
Yes, the server has 2 problems with lists.  
- With only one elements it returns the single element, not an array with 1 element
- With 0 elements, it returns an empty string

There is an issue [open](https://github.com/geosolutions-it/geostore/issues/56) for a lot of time, but wasn't fixed yet.

This issues should be work-arounded anyway at GeoStoreAPI level, or I see that it is solved in `getContextNames` (line 249). 
```
const maps = isArray(results) ? results : (results === "" ? [] : [results]);
```